src/net.h

```diff
@@ -1002,6 +1004,44 @@ class CConnman
     mapMsgTypeSize m_msgtype_bytes_recv;
     mapMsgTypeSize m_msgtype_bytes_sent;
 
+    struct NetMsgStatsKey {
+        // protocol.cpp has a list of allNetMessageTypes which use the NetMsgType namespace.
+        // Can I treat NetMsgType like an enum?
+        std::string msg_type;
+        ConnectionType conn_type;
+        Network net_type;
+
+        // Is this correct? Is there a better way to write this?
+        bool operator< (const NetMsgStatsKey& rhs) const{
+            if (msg_type < rhs.msg_type) {
+                return true;
+            } else if (msg_type == rhs.msg_type) {
+                if (conn_type < rhs.conn_type) {
+                    return true;
+                } else if (conn_type == rhs.conn_type) {
+                    if (net_type < rhs.net_type) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        bool operator= (const NetMsgStatsKey& other) const {
```

Comment:
Overload `==`? Or just get rid of this?

Reply:
Removing the overloaded `<` operator resulted in a bunch of compiler errors, even when I corrected this method to overload `==` (instead of `=`). I'm going to remove the `==` overload method.

Reply:
some context for why `operator<` is relevant - `std::map` is implemented as a balanced tree (usually red-black tree if you remember that from college cs 😛). so it needs to have the comparison defined to be able to place values on the tree

this allows for the logarithmic insertion / search / deletion operations.

Reply:
Had a spare chunk of time that was too small for coding, but just right to catch up on this comment :) Thanks for the resources and the explanation that `std::map` is a self balancing tree. It makes a lot of sense and explains why overloading the `<` operator is necessary!